src/vm/analysis/type_checker/tests/mod.rs

```diff
@@ -1328,6 +1328,60 @@ fn test_buffer_to_ints() {
     }
 }
 
+#[test]
+fn test_string_to_ints() {
+    let good = [
+        r#"(int-to-ascii 1)"#,
+        r#"(int-to-ascii u1)"#,
+        r#"(int-to-utf8 1)"#,
+        r#"(int-to-utf8 u1)"#,
+        r#"(string-to-int "1")"#,
+        r#"(string-to-int u"1")"#,
+        r#"(string-to-uint "1")"#,
+        r#"(string-to-uint u"1")"#,
+    ];
+
+    let expected = [
+        "(string-ascii 1048576)",
+        "(string-ascii 1048576)",
+        "(string-utf8 1048576)",
+        "(string-utf8 1048576)",
+        "int",
+        "int",
+        "uint",
+        "uint",
+    ];
+
+    let bad = [
+        r#"(int-to-ascii 0x0001 0x0001)"#,
+        r#"(int-to-ascii)"#,
+        r#"(int-to-ascii 0x000102030405060708090a0b0c0d0e0f00)"#,
+        r#"(int-to-ascii "a")"#,
```

Comment:
Can you add coverage for `int-to-utf8`?

Reply:
i added fuller coverage for all four functions